corehq/apps/linked_domain/updates.py

```diff
@@ -415,40 +415,46 @@ def update_hmac_callout_settings(domain_link):
 
 def update_auto_update_rules(domain_link):
     if domain_link.is_remote:
-        master_results = remote_get_auto_update_rules(domain_link)
+        upstream_rules = remote_get_auto_update_rules(domain_link)
     else:
-        master_results = local_get_auto_update_rules(domain_link.master_domain)
+        upstream_rules = local_get_auto_update_rules(domain_link.master_domain)
 
-    local_rules = AutomaticUpdateRule.by_domain(
+    downstream_rules = AutomaticUpdateRule.by_domain(
         domain_link.linked_domain,
         AutomaticUpdateRule.WORKFLOW_CASE_UPDATE,
         active_only=False
     )
 
-    # Largely from data_interfacees/models.py - hard_delete()
-    def _delete_rule(rule):
-        rule.delete_criteria()
-        rule.delete_actions()
-        CaseRuleSubmission.objects.filter(rule=rule).delete()
-        rule.delete()
+    for upstream_rule_def in upstream_rules:
+        # Grab local rule by upstream ID (preferred) or by name
```

Comment:
Optional suggestion: it's more common in our codebase to use `get` and a try/except to grab an object that may not be present, like [this](https://github.com/dimagi/commcare-hq/blob/0ae3ec0c52ff19ab82f05b68b6bf85e5fde3e3dd/corehq/apps/case_importer/do_import.py#L612-L615).

Reply:
Funny, I saw that solution proposed on StackOverflow! Felt a bit odd to me but since it's common I'll make the change.

Reply:
Funny, I saw that solution proposed on StackOverflow! Felt a bit odd to me but since it's common I'll make the change.

Reply:
Yeah, a try/except where the exception case is a relatively "normal" workflow is unusual, but this is one pattern that uses it.